elements/lisk-p2p/src/peer/base.ts

```diff
@@ -448,15 +438,26 @@ export class Peer extends EventEmitter {
 		this._peerInfo.internalState.wsMessageRate =
 			(this.peerInfo.internalState.wsMessageCount * RATE_NORMALIZATION_FACTOR) /
 			this._rateInterval;
+
 		this._peerInfo.internalState.wsMessageCount = 0;
 
 		if (
 			this.peerInfo.internalState.wsMessageRate >
-			this._peerConfig.wsMaxMessageRate
+				this._peerConfig.wsMaxMessageRate ||
+			this._protocolRCPCounter > PROTOCOL_REMOTE_EVENT_RCP_EVENTS.size
 		) {
-			this.applyPenalty(this._peerConfig.wsMaxMessageRatePenalty);
+			// Allow to increase penalty based on message rate limit exceeded
+			const messageRateExceedCoeff = Math.floor(
```

Comment:
Avoid abbreviations in variable names. I guess you are referring to `Coefficient`  here as `messageRateExceed Coefficient`